demo/clusters/kind/scripts/common.sh

```diff
@@ -38,6 +38,7 @@ DRIVER_IMAGE_VERSION=$(from_versions_mk "VERSION")
 # The kubernetes tag to build the kind cluster from
 # From https://github.com/kubernetes/kubernetes/tags
 # See also https://hub.docker.com/r/kindest/node/tags
+: ${KIND_K8S_REPO:="https://github.com/kubernetes/kubernetes.git"}
```

Comment:
When I tried this locally, there were issues with extracting the version as below since we're doing a shallow-clone. Were you able to get around this?

Reply:
What do you mean exactly by "issues with extracting the version as below"?

I didn't have any issues. The only place this variable is even used is in `build-kind-image.sh`, so now this just allows what used to be hard-coded there to be overwritten.

Reply:
Let's say that you want to check out a branch:
```
KIND_K8S_REPO=https://github.com/pohly/kubernetes \
KIND_K8S_TAG=dra-structured-parameters \
    ./demo/clusters/kind/create-cluster.sh
```
Since we do a shallow clone of the repo and this does not contain tag information the kubernets build fails.

I was able to get around this locally in initial testing by setting `KUBE_GIT_VERSION=v1.30.0` in addition to the envvars above. This could also be addressed by:
1. Building a full repo clone that's already checked out instead.
2. "unshallowing" the clone so that `git describe --tags` returns some sane version string in the repo.

Reply:
I am also setting `KUBE_GIT_VERSION=v1.30.0` (because this is what you told me to do in slack). I never tried without it (nor really attempted to understand why it was necessary, apparently).